src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -667,11 +667,38 @@ private SourceFeature addSourceQualifiers( TaxonHelper taxonHelper, SampleInfo s
 			select_sourcequalifiers_pstmt = connection.prepareStatement(select_sourcefeature_Query);
 			select_sourcequalifiers_pstmt.setString(1, sampleInfo.getSampleId());
 			select_sourcequalifers_rs = select_sourcequalifiers_pstmt.executeQuery();
+			String latitude = null;
+			String longitude = null;
+			String country = null;
 			while (select_sourcequalifers_rs.next()) {
 				String tag = select_sourcequalifers_rs.getString(1);
 				String value = select_sourcequalifers_rs.getString(2);
+				if (sourceUtils.isCovidTaxId(sourceFeature.getTaxId()) && tag != null) {
+					if (tag.toLowerCase().contains("latitude")) {
+						latitude = value;
+						continue;
+					}
+					if (tag.toLowerCase().contains("longitude")) {
+						longitude = value;
+						continue;
+					}
+					if (tag.trim().equalsIgnoreCase("geographic location (country and/or sea)")) {
+						country = country == null ? value : value +":"+country;
+						continue;
+					}
+					if (tag.trim().equalsIgnoreCase("geographic location (region and locality)")) {
+						country = country == null ? value : country +":"+value;
+						continue;
+					}
+				}
 				sourceUtils.addSourceQualifier(tag, value, sourceFeature);
```

Comment:
I think we should have
```
else {
    sourceUtils.addSourceQualifier(tag, value, sourceFeature);
}
```
i.e. call this only If we are not dealing with master source qualifier values created from multiple sample fields in the above 'if' and 'else if' statements.